packages/core/lib/migrate.ts

```diff
@@ -89,6 +107,55 @@ const migrations: Migration[] = [
       delete updated.data.zones?.[zoneCompound];
     });
 
+    // Migrate zones created by dynamic arrays
```

Comment:
Can we check if `migrationOptions?.migrateDynamicZonesForComponent` exists before doing this work?